Lib/test/_test_multiprocessing.py

```diff
@@ -4472,17 +4473,19 @@ def check_semaphore_tracker_death(self, signum, should_die):
         # bpo-31310: if the semaphore tracker process has died, it should
         # be restarted implicitly.
         from multiprocessing.semaphore_tracker import _semaphore_tracker
-        _semaphore_tracker.ensure_running()
         pid = _semaphore_tracker._pid
+        if pid:
+            os.kill(pid, signal.SIGKILL)
+            time.sleep(0.5)  # give it time to die
```

Comment:
Would it be possible to wait until the process completes instead of using a sleep?

Reply:
Sadly no AFAIK, because the tracker relies on calling `waitpid` to check the child. If you add a `os.waitpid` here you will get:

```traceback

ERROR: test_semaphore_tracker_sigint (test.test_multiprocessing_fork.TestSemaphoreTracker)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pgalindo3/cpython/Lib/test/_test_multiprocessing.py", line 4509, in test_semaphore_tracker_sigint
    self.check_semaphore_tracker_death(signal.SIGINT, False)
  File "/Users/pgalindo3/cpython/Lib/test/_test_multiprocessing.py", line 4481, in check_semaphore_tracker_death
    _semaphore_tracker.ensure_running()
  File "/Users/pgalindo3/cpython/Lib/multiprocessing/semaphore_tracker.py", line 48, in ensure_running
    pid, status = os.waitpid(self._pid, os.WNOHANG)
ChildProcessError: [Errno 10] No child processes
Ran 1 test in 0.068s
```

Reply:
Why not using os.waitpid() in the test, and modify ensure_running() to handle ChildProcessError? The purpose of the waitpid() call in ensure_running() is to check if the process died. We don't care of the exit status.

Adding time.sleep(0.5) means adding a new race condition, while fixing another, no?